### ISSUE TITLE
Rule0044AnalyzeTransferFields (Prerelease?)

### ISSUE THREAD

**Issue by NAVFreak** [StefanMaron/BusinessCentral.LinterCop#431]
There seems to have been an update yesterday evening with some disturbances.
I haven't done a nr to text translation yet so I can't currently see the exact version in the pipeline
https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/download/v0.30.***0/BusinessCentral.LinterCop.current.dll

BC version 22.5

This morning it seems to have boiled down to:
```
##[warning]warning AD000***: Analyzer 'BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields' threw an exception of type 'System.InvalidCastException' with message 'System.InvalidCastException: Unable to cast object of type 'Microsoft.Dynamics.Nav.CodeAnalysis.Syntax.FieldModificationSyntax' to type 'Microsoft.Dynamics.Nav.CodeAnalysis.Syntax.FieldSyntax'.
##[warning]at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.Table.PopulateFields(FieldExtensionListSyntax fieldList)
##[warning]at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.GetTableWithFieldsByTableName(Compilation compilation, String tableName, Dictionary`*** tableExtensions)
##[warning]at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.AnalyzeTableExtension(SyntaxNodeAnalysisContext ctx)
##[warning]at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass5***_***.<ExecuteSyntaxNodeAction>b__***() in D:\a\_work\***\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 745
##[warning]at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`*** info) in D:\a\_work\***\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line ***086'_

```
Looks like the transferfield rule I requested 😁
Can I help with troubleshooting? If this is a bug that is derived from source repo it should also be fixed there, right @tinestaric ?

**Comment by tinestaric** [comment]
Huh, apparently in the few months we've been running the rule, we've never added a modify(FieldName) to our table extensions, to break it... :D

I'd say, let's fix it here, and I'll backport the fix to CompanialCop later, as the rules are not 1-to-1 copies.

**Comment by Arthurvdv** [comment]
@tinestaric pinged my this morning about this issue and could point me exactly to where I needed to look to resolved this. That was very helpful.

So I assistent on creating an PR for this, where this now should be resolved in the updated (pre)release version [v0.30.10](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.10).

**Comment by fvet** [comment]
Facing a similar warning

```
Analyzer 'BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields' threw an exception of type 'System.NullReferenceException' with message 'System.NullReferenceException: Object reference not set to an instance of an object.
   at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.Table.PopulateFields(FieldExtensionListSyntax fieldList)
   at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.AnalyzeTableExtension(SyntaxNodeAnalysisContext ctx)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass52_1.<ExecuteSyntaxNodeAction>b__1() in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 745
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1102'

```

```
Analyzer 'BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields' threw an exception of type 'System.NullReferenceException' with message 'System.NullReferenceException: Object reference not set to an instance of an object.
   at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.Table.PopulateFields(FieldExtensionListSyntax fieldList)
   at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.GetTableWithFieldsByTableName(Compilation compilation, String tableName, Dictionary`2 tableExtensions)
   at BusinessCentral.LinterCop.Design.Rule0044AnalyzeTransferFields.AnalyzeTableExtension(SyntaxNodeAnalysisContext ctx)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass52_1.<ExecuteSyntaxNodeAction>b__1() in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 745
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info) in D:\a\_work\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1102'
```

**Comment by tinestaric** [comment]
@Arthurvdv I'm guessing a tableextension, that doesn't have any fields is what could be the cause.

**Comment by NAVFreak** [comment]
> @Arthurvdv I'm guessing a tableextension, that doesn't have any fields is what could be the cause.

Which can happen if you want to extend a table with new methods. 
For instance. Customer.<prefix>_GetCustomerCreditLimit

**Comment by Arthurvdv** [comment]
> I'm guessing a tableextension, that doesn't have any fields is what could be the cause.

Spot on, that was indeed the case it seems.

The `System.NullReferenceException` exceptions should now be resolved in the updated (pre)release version [v0.30.10](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.30.10) release.

**Comment by NAVFreak** [comment]
It is working for us now, and no transferfield warnings (in this app) ✌️
